fairscale/nn/pipe/messages.py

```diff
@@ -0,0 +1,137 @@
+import pickle
+from queue import Empty as QueueEmpty
+from queue import Queue
+from typing import Any, List
+
+import numpy as np
+import torch
+from torch import Tensor
+
+from fairscale.nn.model_parallel import get_pipeline_parallel_group
+
+from .types import MESSAGE_GENERATION_START, InputDevice, PipeMessage, Tensors, TransportConfig
+
+# FIXME Why is 256 ok for training but not for tests?
+MESSAGE_TENSOR_SIZE = 1024  # 256
```

Comment:
Why does this need to be fixed-size?

Reply:
Because send/recv have to agree on the size, so the two options are send the size first, or just pre-allocate a large enough buffer for all cases, so I chose the latter